doc/jinja-filters.md

```diff
@@ -0,0 +1,438 @@
+
+##### Parameters: 
+- from_str: datetime string of earliest date.
+- to_str: datetime string of later date. If set to 'now' will use today's date.
+
+##### Context:
+Used in Census to work out a person's age from a given date of birth:
+```
+You are {{ calculate_years_difference (answers['date-of-birth-answer'][group_instance], 'now') }} old. Is this correct?
+
+As an option:
+Yes, I am {{ calculate_years_difference (answers['date-of-birth-answer'][group_instance], 'now') }} old
+```
+
+##### Examples:
+```
+Input:                          Output:
+("2017-01-30", "2018-01-30')    "1 year"
+("2015-02-02", "2018-02-01")    "2 years"
+("2016-02-29", "2017-02-28")    "1 year"
+("2016-02-29", "2020-02-28")    "3 years"
+```
+
+## format_date_range_no_repeated_month_year
+Format a date range, ensuring months and years are not repeated.
+If the dates are in the same year, the first year (YYYY) will be removed.
+If the dates are in the same month and year, the first year (YYYY) and month will be removed.
+
+##### Parameters: 
+- start_date: Initial date in range.
+- end_date: Final date in range.
+- date_format: Format to return date in. Default is 'd MMMM YYYY'.
+
+##### Context:
+Used in LMS to simplify dates when often they are only a week apart.
+```
+Did you have a paid job, either as an employee or self-employed, in the week {{ format_date_range_no_repeated_month_year(calculate_offset_from_weekday_in_last_whole_week(collection_metadata['started_at'], {}), calculate_offset_from_weekday_in_last_whole_week(collection_metadata['started_at'], {}, 'SU'), 'EEEE d MMMM YYYY') }}?
+```
+
+##### Examples:
+```
+Input:                                              Output:
+("2018-08-14", "2018-08-16", "EEEE d MMMM YYYY")    "<span class='date'>Tuesday 14</span> to <span class='date'>Thursday 16 August 2018</span>"
+("2018-07-31", "2018-08-16", "EEEE d MMMM YYYY")    "<span class='date'>Tuesday 31 July</span> to <span class='date'>Thursday 16 August 2018</span>"
+("2017-12-31", "2018-08-16", "EEEE d MMMM YYYY")    "<span class='date'>Sunday 31 December 2017</span> to <span class='date'>Thursday 16 August 2018</span>"
+("2017-12-31", "2018-08-16", "MMMM YYYY")           "<span class='date'>December 2017</span> to <span class='date'>August 2018</span>"
+("2018-08-14", "2018-08-16", "MMMM YYYY")           "<span class='date'>August 2018</span> to <span class='date'>August 2018</span>"
+("2017-12-31", "2018-08-16", "YYYY")                "<span class='date'>2017</span> to <span class='date'>2018</span>"
+("2017-07-31", "2018-08-16", "YYYY")                "<span class='date'>2017</span> to <span class='date'>2018</span>"
+("2018-08-14", "2018-08-16", "EEEE d")              "<span class='date'>Tuesday 14</span> to <span class='date'>Thursday 16</span>"
+```
+
+
+## first_non_empty_item
+Returns the first non empty value in a list of values.
+
+##### Parameters: 
+- *items: List of items where some of them could have no value.
```

Comment:
Not sure but `List of items` to me sounds like 
``` python 
first_non_empty_item([1,'',3])
# returns [1, '', 3]
```
instead of
``` python
first_non_empty_item(1, '', 3)
# returns 1
```
Maybe `1 or more arguments`?
🤷‍♂️